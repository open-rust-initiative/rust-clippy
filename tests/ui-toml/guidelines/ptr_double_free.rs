#![allow(unused)]
#![warn(clippy::ptr_double_free)]

fn my_free(_ptr: *mut i8) {}

fn main() {
    let ptr: *mut i8 = 1 as *mut _;

    my_free(ptr);
    my_free(ptr); // lint

    let ptr1: *mut i8 = 1 as *mut _;
    let ptr2: *mut i8 = 1 as *mut _;
    let ptr3: *mut i8 = 1 as *mut _;
    my_free(ptr1);
    my_free(ptr2);
    my_free(ptr1); // lint
    my_free(ptr3);
}
