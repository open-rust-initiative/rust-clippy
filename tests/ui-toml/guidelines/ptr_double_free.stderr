error: pointer was freed multiple times
  --> $DIR/ptr_double_free.rs:9:5
   |
LL |     my_free(ptr);
   |     ^^^^^^^^^^^^
   |
note: second free occurred here
  --> $DIR/ptr_double_free.rs:10:5
   |
LL |     my_free(ptr); // lint
   |     ^^^^^^^^^^^^
   = note: `-D clippy::ptr-double-free` implied by `-D warnings`

error: pointer was freed multiple times
  --> $DIR/ptr_double_free.rs:15:5
   |
LL |     my_free(ptr1);
   |     ^^^^^^^^^^^^^
   |
note: second free occurred here
  --> $DIR/ptr_double_free.rs:17:5
   |
LL |     my_free(ptr1); // lint
   |     ^^^^^^^^^^^^^

error: aborting due to 2 previous errors

